src/Dialogs/FrequencyDialog.cpp

```diff
@@ -167,24 +179,18 @@ FrequencyListWidget::UpdateList() noexcept
 void
 FrequencyDialogShowModal() noexcept
 {
-  static std::vector<FrequencyListRenderer::RadioChannel> channels;
+  static std::vector<FrequencyListWidget::RadioChannel> channels;
```

Comment:
Another global variable

Reply:
This is a pattern I borrowed from elsewhere in the codebase (eg dlgChecklist), but as you've pointed out separately there's quite a bit of legacy code. What would be the preferred way of persisting an object (in this case the collection of channels, but my questions's more general) between invocations of the dialog?

Reply:
The checklist dialog is a special example of bad and old code which I havn't rewritten, because I never cared about this feature.
You can't imagine how many global variables I have removed in the last 13 years. Hundreds? Or thousands? I can't remember. XCSoar used to be a VERY horrible mess of intertwined global state. It's still messy, but it used to be much worse. My task is to keep new bad code out, and try to reduce the amount of existing bad code.
There still isn't a good "one-fits-all" solution for global state; for example there is Components.hpp which is a huge unholy collection of global variables which havn't found a proper home somewhere else. But at least they get initialised and deinitialised properly. (More or less; if the XCSoar process gets recycled by the Android OS, and the BSS section doesn't get reinitialised, it gets dirty... sometimes, I wish I didn't know and care about all that, then I could sleep better. I could talk for days about how horrible real-life code is.)

In your case, I'd say: what's the point of persisting this object? You want to load it only once. But for now (because we don't have a proper solution to add global state for random one-off features), just load the file every time the dialog opens. It's a kilobyte or so, probably less. Don't bother caching it. For now.
Of course, in the long run, it should be loaded only once, and XCSoar should access the file system only when necessary. But we're not there yet.